### PR TITLE
Convert to ES6

### PR DESCRIPTION
**Pull request by blikblum**
Converts to ES6 syntax, removing dependency on coffescript

Tests also converted

### CODE REVIEW

File: src/generate_data.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      type = rangeType;
+    }
+    
+    end = rangeEnd;
+  }
+  
+  trie.setRange(parseInt(start, 16), parseInt(end, 16), classes[type], true);
+
+  // write the trie to a file
+  fs.writeFile(__dirname + '/classes.trie', trie.toBuffer());
```
Comment (liborm85) [inline src/generate_data.js:47]: `writeFile` requires callback function for compatibility with nodejs 10+ ([see](https://nodejs.org/api/fs.html#fs_fs_writefile_file_data_options_callback)). Better is use instead `writeFileSync` method.

Comment (blikblum) [reply src/generate_data.js:47]: This is just a port to ES6. Further changes must be done on a separated PR

File: src/generate_data.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const fs = require('fs');
+const request = require('request');
+const classes = require('./classes');
+const UnicodeTrieBuilder = require('unicode-trie/builder');
+
+// this loads the LineBreak.txt file for Unicode and parses it to
+// combine ranges and generate CoffeeScript
```
Comment (devongovett) [inline src/generate_data.js:7]: coffeescript -> javascript

**Review by devongovett — APPROVED** [review]

**Comment by devongovett** [comment]
Bumped to v1.0.0 and published.

**Pull request merged**